src/modding/ModHelpers.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections.Generic;
+using Godot;
+using static FullModDetails;
+
+public static class ModHelpers
+{
+    /// <summary>
+    ///   Checks if the mod is Compatible With Thrive
+    /// </summary>
+    /// <returns> Returns true if the mod is compatible with this version of thrive or if is unknown. </returns>
+    public static bool IsCompatible(this VersionCompatibility compatibleVersions)
+    {
+        return compatibleVersions > VersionCompatibility.NotExplicitlyCompatible;
+    }
+
+    /// <summary>
+    ///   Turns the result from a check into a string of the error and how to fix it
+    /// </summary>
+    /// <returns> Returns a translated string of the possible error. </returns>
+    public static string CheckResultToString((int ErrorType, int ModIndex, int OtherModIndex) checkResult,
+        List<FullModDetails> list)
+    {
+        var result = string.Empty;
+
+        // The mod that is causing the error
+        ModInfo offendingMod = new ModInfo();
+        if (checkResult.ModIndex >= 0)
+        {
+            offendingMod = list[checkResult.ModIndex].Info;
+        }
+        else
+        {
+            offendingMod.Name = TranslationServer.Translate("UNKNOWN_MOD");
+        }
+
+        // The reason why the mod is causing an error
+        ModInfo otherMod = new ModInfo();
+        if (checkResult.OtherModIndex >= 0)
+        {
+            otherMod = list[checkResult.OtherModIndex].Info;
+        }
+        else
+        {
+            otherMod.Name = TranslationServer.Translate("UNKNOWN_MOD");
```

Comment:
Here again it would be nice to have the specific mod error class so that the name of the problematic mod could be used as a fallback before ultimately showing unknown mod if nothing is known about it.